### ISSUE TITLE
hadolint gives irrelevant shellcheck warnings with heredoc

### ISSUE THREAD

**Issue by rittneje** [hadolint/hadolint#806]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Hadolint should not give false warnings about `pipefail` when used with a heredoc that specifies a shebang.

### Actual behavior

```
-:4 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
-:4 SC3040 warning: In POSIX sh, set option pipefail is undefined.
```

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v2.9.3-0-g346e419-dirty
```

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker.io/docker/dockerfile:labs

FROM ubuntu:focal
RUN <<EOF
#!/usr/bin/env bash
set -eux -o pipefail
echo 'abc' | cat
EOF
```

**Additional environment details (OS, stack version, etc.)**

**Comment by rittneje** [comment]
It seems that with `RUN bash <<EOF` it no longer recognizes it as shell commands so no linting at all is performed on the heredoc. That is another bug that should be fixed.

I also tested with `#!/bin/bash` and that too leads to false warnings.

**Comment by felipecrs** [comment]
These forms also produce issues:

```
❯ hadolint <(printf '%s\n' 'FROM ubuntu:24.04' 'RUN <<EOF bash' 'echo "${var//1/2}"' 'EOF')
/proc/self/fd/18:2 SC3060 warning: In POSIX sh, string replacement is undefined.

❯ hadolint <(printf '%s\n' 'FROM ubuntu:24.04' 'RUN <<EOF' '#!/bin/bash' 'echo "${var//1/2}"' 'EOF')
/proc/self/fd/18:2 SC3060 warning: In POSIX sh, string replacement is undefined.
```
